content/articles/task-tracker-app-with-react.js-using-hooks/index.md

```diff
@@ -0,0 +1,231 @@
+Task tracker app with React hooks
+
+##Introduction
+
+In our everyday lives, we all have tasks we plan on completing before the day runs out. Often, we use task tracking applications to keep track of everything we have to do.
+
+In this article, you will learn how to build a task tracker with React that allows you to add, retrieve, update, and delete tasks. You will also learn how to use hooks, manage states and make use of props in React.
+
+##Getting started
+
+First, create a react app; here is a link to learn how to do that <https://reactjs.org/docs/create-a-new-react-app.html>.
+
+After completing your react app and setting up your code editor, we create our component folder. Our app will consist of three main components, which are TodoForm.js,TodoList.js, Todo.js.
+
+All our functionality will be in our App.js, which will serve as an entry point for our app. For styling purposes, we will use material UI instead of our regular CSS. Visit material UI to install material UI<https://mui.com/getting-started/installation/> if you don’t have it installed already.
+
+##TodoForm
+
+TodoForm will be responsible for rendering the form necessary for us to add a to-do to the list. Using the short form, which is import react and export TodoForm, define a function called TodoForm.
+
+In this TodoForm function, go ahead and return a form element with an input and a button inside of it. Inside the TodoForm, you need to import useState because we need to define some state to keep track of input from the user.
+
+Inside the TodoForm function, you need to define a state called todo, setTodo as the function initialize as an object with three properties, id which is a string, task which is also a string, and completed which is a boolean. Completed will be used to track whether or not the Todo is marked as completed, and the task will describe the Todo.
+
+We need to define a function for when a user types in input for a Todo to keep track of it in our state. We will create a function called handleTaskInputChange that takes in an event as the parameter. This function will be responsible for updating the task property on our Todo object.
+
+In this function, we will call setTodo and pass in a new object with the old Todo property and update the task property with the new value from the event target value. In this case, the event target value contains the new input text from the user.
+
+Our return statement will define the onChange function to run our new handleTaskInputChange function every time the event fires. In this case, onChange will fire every time the input value changes. Also, we set the input value to be todo task because that’s what is updated every time the handleTaskInputChange is called. We also give our button a type called to submit. So now that we have some input data to work with, we need to handle the case when we want to add a new todo to the list. 
```

Comment:
Ensure all non-English words such as functions standout properly. For example, instead of ...'onChange' function .....write `onChange()` function.